### ISSUE TITLE
Data list doesn't use semantic heading element

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-ui-kit#3884]
### Summary of the bug 
The data list component has a heading which is styled as a level 3 heading, but it doesn't actually use an `<h3>` element. 

### 🪜 How to reproduce 
1. Go to the [data list component page](https://design.sis.gov.uk/components/data-list/code).
2. Inspect the heading on a data list.
3. See that there's no `<h3>` in the HTML.

### 📸 Screenshots or code 

<img width="350" height="241" alt="Image" src="https://github.com/user-attachments/assets/3cb87f5e-9f6c-4997-8b7a-1cd6ad79cac2" />

### 🧐 Expected behaviour 
I think there should be an actual `<h3>` within the `<ic-typography>`. (Not sure if we decided not to add one in case it causes conflicts in the heading level order, if there were other `<h3>`s on a page)

### 🚨 Urgency (low, medium or high)
Probably medium urgency, as its accessibility-related but there is a workaround.

### Additional info 
This was noticed by a customer. However, I let them know that they could use the "heading" slot to add their own `<h3>`.

This may require updates to the data list examples on the ICDS website, because the section headings use `<h3>`s and `<h4>`s. May have to use slots for all the examples, to use `<h5>`s?

**Comment by ad9242** [comment]
closing as slot can be used instead and this gives control over heading level
